### PR TITLE
In spec_augment.py, check whether an array is writeable before modifying it inplace

### PR DESCRIPTION
**Pull request by mdecerbo**
This fixes occasional, unpredictable "ValueError: assignment destination is read-only" crashes for me. See #5415 .

### CODE REVIEW

**Comment by sw005320** [comment]
@pyf98, do you think it is OK?

File: espnet/transform/spec_augment.py
Comment (pyf98) [inline espnet/transform/spec_augment.py:1]: It looks good to me (although I have never encountered this issue before).

**Review by pyf98 — APPROVED** [review]

**Comment by sw005320** [comment]
Thanks, @mdecerbo and @pyf98!

**Comment by mdecerbo** [comment]
Thanks to you @pyf98 and @sw005320 for merging the patch, now I don't have to keep remembering to make that change each time I update my espnet checkout! 🙂

**Pull request merged**